File: app/src/test/java/com/mgaetan89/showsrage/fragment/ShowsFragment_OnQueryTextSubmitTest.java
package com.mgaetan89.showsrage.fragment;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.junit.Assert.assertTrue;
import static org.mockito.Mockito.spy;
import static org.mockito.Mockito.verify;

@RunWith(Parameterized.class)
public class ShowsFragment_OnQueryTextSubmitTest {
	@Parameterized.Parameter(0)
	public String query;

	private ShowsFragment fragment;

	@Before
	public void before() {
		this.fragment = spy(new ShowsFragment());
	}

	@Test
	public void onQueryTextSubmit() {
		try {
			assertTrue(this.fragment.onQueryTextSubmit(this.query));
			verify(this.fragment).sendFilterMessage();
		} catch (NullPointerException exception) {
			// LocalBroadcastManager.getInstance(Context) returns null in tests
		}
	}

	@After
	public void after() {
		this.fragment = null;
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{null},
				{""},
				{" "},
				{"Search Query"},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/ShowsFragment_OnQueryTextSubmitTest.kt
package com.mgaetan89.showsrage.fragment

import org.junit.Assert.assertTrue
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized
import org.mockito.Mockito.spy
import org.mockito.Mockito.verify

@RunWith(Parameterized::class)
class ShowsFragment_OnQueryTextSubmitTest(val query: String?) {
    private lateinit var fragment: ShowsFragment

    @Before
    fun before() {
        this.fragment = spy(ShowsFragment())
    }

    @Test
    fun onQueryTextSubmit() {
        try {
            assertTrue(this.fragment.onQueryTextSubmit(this.query))
            verify<ShowsFragment>(this.fragment).sendFilterMessage()
        } catch (exception: NullPointerException) {
            // LocalBroadcastManager.getInstance(Context) returns null in tests
        }

    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any?>> {
            return listOf(
                    arrayOf<Any?>(null),
                    arrayOf<Any?>(""),
                    arrayOf<Any?>(" "),
                    arrayOf<Any?>("Search Query")
            )
        }
    }
}
